test_conformance/api/test_queue_properties_queries.cpp

```diff
@@ -0,0 +1,313 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "testBase.h"
+#include "harness/typeWrappers.h"
+#include <vector>
+#include <algorithm>
+
+struct test_queue_array_properties_data
+{
+    std::vector<cl_queue_properties> properties;
+    std::string description;
+};
+
+int verify_if_properties_supported(
+    cl_device_id deviceID, cl_command_queue_properties requested_bitfield,
+    cl_uint requested_size)
+{
+    int error = CL_SUCCESS;
+    bool on_host_queue = true;
+
+    if ((requested_bitfield & CL_QUEUE_ON_DEVICE)
+        || (requested_bitfield & CL_QUEUE_ON_DEVICE_DEFAULT))
+    {
+        on_host_queue = false;
+    }
+    cl_command_queue_properties supported_properties = 0;
+    cl_command_queue_properties all_properties = 0;
+
+    std::vector<cl_command_queue_properties> all_properties_vector{
+        CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, CL_QUEUE_PROFILING_ENABLE
+    };
+    for (auto each_property : all_properties_vector)
+    {
+        all_properties |= each_property;
+    }
+    cl_command_queue_properties requested_properties =
+        all_properties & requested_bitfield;
+
+    if (on_host_queue)
+    {
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_HOST_PROPERTIES,
+                                sizeof(supported_properties),
+                                &supported_properties, NULL);
+        test_error(error,
+                   "clGetDeviceInfo asking for "
+                   "CL_DEVICE_QUEUE_ON_HOST_PROPERTIES failed");
+    }
+    else
+    {
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_DEVICE_PROPERTIES,
+                                sizeof(supported_properties),
+                                &supported_properties, NULL);
+        test_error(error,
+                   "clGetDeviceInfo asking for "
+                   "CL_DEVICE_QUEUE_ON_DEVICE_PROPERTIES failed");
+        if (supported_properties == 0)
+        {
+            log_info("\tCL_DEVICE_QUEUE_SUPPORTED false, skipped\n");
+            return TEST_SKIPPED_ITSELF;
+        }
+        else if (requested_size > 0)
+        {
+            cl_uint max_packet_size = 0;
```

Comment:
```suggestion
            cl_uint max_size = 0;
```

Copy/paste from pipes? :)

Reply:
Hah, could be!  Fixed.